Add beta -> upgrade docs (#2407)
elastic/cloud-on-k8s: docs/release-notes/highlights-1.0.0-beta1.asciidoc

=== Breaking Changes

The 1.0.0-beta version of the operator does not delete resources created by older versions of the operator, but it also does not manage them. 

> Attempting to delete resources created with a 0.9.0 version will hang if ECK 1.0.0-beta1 is running. Please see the <<{p}-upgrading-eck, upgrading documentation>> for more information.

Reviewer: ```suggestion
You might not be able to delete resources created with a 0.9.0 version if ECK 1.0.0-beta1 is running. See the <<{p}-upgrading-eck, upgrading documentation>> for more information.
```
